src/main/java/io/pinecone/PineconeConfigs.java

```diff
@@ -0,0 +1,53 @@
+package io.pinecone;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PineconeConfigs {
+    private PineconeClientConfig clientConfig;
+    private PineconeConnectionConfig connectionConfig;
+
+    public PineconeConfigs() {
+        clientConfig = new PineconeClientConfig();
+        connectionConfig = new PineconeConnectionConfig();
+    }
+
+    public PineconeConfigs(PineconeClientConfig clientConfig, PineconeConnectionConfig connectionConfig) {
+        this.clientConfig = clientConfig;
+        this.connectionConfig = connectionConfig;
+    }
+
+    public PineconeConfigs(String apiKey, String connectionURL) {
+        this();
+        String pattern = "https://([a-zA-Z0-9-]{1,45})-([a-fA-F0-9]+)\\.svc\\.([a-zA-Z0-9]+)\\.pinecone\\.io";
+        Pattern regexPattern = Pattern.compile(pattern);
+        Matcher matcher = regexPattern.matcher(connectionURL);
+
+        if (matcher.matches()) {
+            String indexName = matcher.group(1);
+            String projectName = matcher.group(2);
+            String environment = matcher.group(3);
+            clientConfig = clientConfig.withApiKey(apiKey).withProjectName(projectName).withEnvironment(environment);
+            connectionConfig = connectionConfig.withIndexName(indexName);
+        }
+        else {
+            throw new PineconeValidationException("Unable to parse connection url");
```

Comment:
Throwing exception if the url wasn't parsable but please feel free to comment on if this user flow doesn't look good. Another option is to return a new PineconeClientConfig with new PineconeClientConfig and PineconeConnectionConfig but the values (apiKey, env, projectName, and indexName) will not be set. So it'll throw invalidException.